source/includes/code-examples/change-streams/change-streams.cs

```diff
@@ -65,6 +65,115 @@ await cursor.ForEachAsync(change =>
+        {
+            changeStreamEnumerator.MoveNext();
+            fragment = changeStreamEnumerator.Current;
+            MergeFragment(changeStreamEvent, fragment);
+        }
+    }
+    return changeStreamEvent;
+}
+
+// Merges a fragment into the base event
+private static void MergeFragment<TDocument>(
+    ChangeStreamDocument<TDocument> changeStreamEvent,
+    ChangeStreamDocument<TDocument> fragment)
+{
+    foreach (var element in fragment.BackingDocument)
+    {
+        if (element.Name != "_id" && element.Name != "splitEvent")
+        {
+            changeStreamEvent.BackingDocument[element.Name] = element.Value; 
+        }
+    }
+}
+// end-split-event-helpers-sync
+
+// start-split-event-helpers-async
+// Fetches the next complete change stream event
+private static async Task<ChangeStreamDocument<TDocument>> GetNextChangeStreamEventAsync<TDocument>(
+    IAsyncCursor<ChangeStreamDocument<TDocument>> changeStreamCursor)
+{
+    var changeStreamEvent = changeStreamCursor.Current.First();
+
+    // Reassembles change event fragments if the event is split
+    if (changeStreamEvent.SplitEvent != null)
+    {
+        var fragment = changeStreamEvent;
+        while (fragment.SplitEvent.Fragment < fragment.SplitEvent.Of)
+        {
+            if (!await changeStreamCursor.MoveNextAsync())
+            {
+                throw new InvalidOperationException("Incomplete split event fragments.");
+            }
+            fragment = changeStreamCursor.Current.First();
+            MergeFragment(changeStreamEvent, fragment);
+        }
+    }
+    return changeStreamEvent;
+}
+
+// Merges a fragment into the base event
+private static void MergeFragment<TDocument>(
+    ChangeStreamDocument<TDocument> changeStreamEvent,
+    ChangeStreamDocument<TDocument> fragment)
+{
+    foreach (var element in fragment.BackingDocument)
+    {
+        if (element.Name != "_id" && element.Name != "splitEvent")
+        {
+            changeStreamEvent.BackingDocument[element.Name] = element.Value;
+        }
+    }
+}
+// end-split-event-helpers-async
+
+// start-split-change-event-async
+var pipeline = new EmptyPipelineDefinition<ChangeStreamDocument<Restaurant>>()
+    .ChangeStreamSplitLargeEvent();
+
+using (var cursor = await collection.WatchAsync(pipeline))
+{
+    while (await cursor.MoveNextAsync())
+    {
+        foreach (var changeStreamEvent in cursor.Current)
+        {
+            var completeEvent = await GetNextChangeStreamEventAsync(cursor);
+            Console.WriteLine("Received the following change: " + completeEvent.BackingDocument);
+        }
+    }
+}
+// end-split-change-event-async
+
+// start-split-change-event-sync
+var pipeline = new EmptyPipelineDefinition<ChangeStreamDocument<Restaurant>>()
+    .ChangeStreamSplitLargeEvent();
+
+using (var cursor = collection.Watch(pipeline))
+{
+    using (var enumerator = cursor.ToEnumerable().GetEnumerator())
```

Comment:
@BorisDog do we need to do all of this here?

Reply:
Short answer: unfortunately yes.
We just demonstrate how to merge the events if the user chooses to. Our spec does not prescribe any merging functionality, so we did not implement a shortcut for that in our API. We also didn't have any requests for that yet.